src/server/JasmineGraphServer.cpp

```diff
@@ -1665,35 +1493,47 @@ static void degreeDistributionCommon(std::string graphID, std::string command) {
-        bcopy((char *)server->h_addr, (char *)&serv_addr.sin_addr.s_addr, server->h_length);
-        serv_addr.sin_port = htons(port);
-        if (Utils::connect_wrapper(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
-            continue;
-        }
+        for (std::vector<std::string>::iterator partitionit = workerPartition.partitionID.begin();
+             partitionit != workerPartition.partitionID.end(); partitionit++) {
+            std::string partition = *partitionit;
+            bzero((char *)&serv_addr, sizeof(serv_addr));
+            serv_addr.sin_family = AF_INET;
+            bcopy((char *)server->h_addr, (char *)&serv_addr.sin_addr.s_addr, server->h_length);
+            serv_addr.sin_port = htons(port);
+            if (Utils::connect_wrapper(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
+                continue;
+            }
 
-        if (!Utils::sendExpectResponse(sockfd, data, INSTANCE_DATA_LENGTH, command, JasmineGraphInstanceProtocol::OK)) {
-            close(sockfd);
-            continue;
-        }
+            if (!Utils::sendExpectResponse(sockfd, data, INSTANCE_DATA_LENGTH, command,
+                                           JasmineGraphInstanceProtocol::OK)) {
+                Utils::send_str_wrapper(sockfd, JasmineGraphInstanceProtocol::CLOSE);
+                close(sockfd);
+                continue;
+            }
 
-        if (!Utils::sendExpectResponse(sockfd, data, INSTANCE_DATA_LENGTH, graphID, JasmineGraphInstanceProtocol::OK)) {
-            close(sockfd);
-            continue;
-        }
+            if (!Utils::sendExpectResponse(sockfd, data, INSTANCE_DATA_LENGTH, graphID,
+                                           JasmineGraphInstanceProtocol::OK)) {
+                Utils::send_str_wrapper(sockfd, JasmineGraphInstanceProtocol::CLOSE);
+                close(sockfd);
+                continue;
+            }
 
-        if (!Utils::sendExpectResponse(sockfd, data, INSTANCE_DATA_LENGTH, partition,
-                                       JasmineGraphInstanceProtocol::OK)) {
-            close(sockfd);
-            continue;
-        }
+            if (!Utils::sendExpectResponse(sockfd, data, INSTANCE_DATA_LENGTH, partition,
+                                           JasmineGraphInstanceProtocol::OK)) {
+                Utils::send_str_wrapper(sockfd, JasmineGraphInstanceProtocol::CLOSE);
+                close(sockfd);
+                continue;
+            }
 
-        if (!Utils::send_str_wrapper(sockfd, workerList)) {
+            if (!Utils::send_str_wrapper(sockfd, workerList)) {
+                Utils::send_str_wrapper(sockfd, JasmineGraphInstanceProtocol::CLOSE);
+                close(sockfd);
+                continue;
+            }
+            Utils::send_str_wrapper(sockfd, JasmineGraphInstanceProtocol::CLOSE);
             close(sockfd);
-            continue;
+            server_logger.info("Sent: " + workerList);
```

Comment:
There is no point of having this logger line here

Reply:
Removed

Reply:
Ok